tests/unit/test_ad_model.py

```
@@ -60,3 +98,220 @@ def is_close(a:float | None, b:float) -> bool:
+def complete_ad_cfg(base_ad_cfg:dict[str, object]) -> dict[str, object]:
+    return base_ad_cfg | {
+        "republication_interval": 7,
+        "price": 100,
+        "auto_reduce_price": True,
+        "price_reduction": {"type": "FIXED", "value": 5},
+        "min_price": 50
+    }
+
+
+class SparseAdDefaults(AdDefaults):
+    def model_dump(self, *args:Any, **kwargs:Any) -> dict[str, object]:
```

Ruff (0.14.4)

Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)